lua/entities/gmod_wire_expression2/base/parser.lua

```diff
@@ -82,6 +82,10 @@ E2Lib.Parser = {}
 local Parser = E2Lib.Parser
 Parser.__index = Parser
 
+local parserDebug = CreateConVar("wire_expression2_parser_debug", 0,
```

Comment:
e2 already has a debug convar, why not reuse that?

Reply:
`wire_expression2_debug` adds runtime typechecking and breaks E2s - if I was investigating a problem with E2, I'd want to turn on one or the other but probably not both. Being able to debug each component of the system individually is helpful.

If you'd like I can replace this with a hardcoded:
```Lua
local parserDebug = false -- set this to true to see the parse tree output
```
I don't like that though - when somebody reports their E2 is misbehaving in a way that sounds like a problem with the parsing (or optimization, respectively) it'd be useful to be able to tell them to turn on this convar and send the results. `wire_expression2_debug` stops lots of E2s from running.

Reply:
oh it breaks e2? Didn't know that

Reply:
My memory might be outdated but I think lots of common constructs (eg. `while`) don't properly specify type information for the the operators (eg. 'op:whl').

Reply:
If wire_expression2_debug breaks an E2 because of the while op, it needs to be updated.

Reply:
It used to break `foreach` though that has since been fixed. It might have had an issue with `while` at some point, but as far as I can tell it no longer does and I can't find any issues of such occurring.

There's #1180, also after a quick look I can find both `A ?: B` and `A ? B : C` fail it. Probably a few other things I'm missing.

Reply:
Yeah, there's a couple of other ops which are eg. homomorphisms, that the compiler doesn't actually know about. There's also function calls and string function calls that the return value of isn't known that mess with `wire_expression2_debug`. I agree that it should be fixed - I'd like to address that properly by adding a type-checking pass between the parser and compiler that does a smarter job than the compiler currently does.